examples/omnivore/data/datasets.py

```diff
@@ -0,0 +1,242 @@
+
+        img_depth = PIL.Image.open(depth_path)
+        tensor_depth = T.ToTensor()(img_depth)
+        tensor_disparity = baseline * focal_length / (tensor_depth / 1000.0)
+        return tensor_disparity
+
+    def _read_sunrgbd_image(self, image_dir):
+        rgb_dir = os.path.join(image_dir, "image")
+        rgb_path = os.path.join(rgb_dir, os.listdir(rgb_dir)[0])
+        img_rgb = PIL.Image.open(rgb_path)
+        tensor_rgb = T.ToTensor()(img_rgb)
+
+        tensor_d = self._get_disparity_tensor(image_dir)
+
+        tensor_rgbd = torch.cat((tensor_rgb, tensor_d), dim=0)
+        return tensor_rgbd
+
+    def _get_sunrgbd_scene_class(self, image_dir):
+        with open(os.path.join(image_dir, "scene.txt"), "r") as fin:
+            scene_class = fin.read().strip()
+        return scene_class
+
+    def __getitem__(self, idx):
+        # return tuple of image (H W C==4) and scene class index
+        image_dir = self.image_dirs[idx]
+        x_rgbd = self._read_sunrgbd_image(image_dir)
+        scene_class = self._get_sunrgbd_scene_class(image_dir)
+        scene_idx = self.class_to_idx[scene_class]
+
+        if self.transform:
+            x_rgbd = self.transform(x_rgbd)
+
+        if self.target_transform:
+            scene_idx = self.target_transform(scene_idx)
+
+        return x_rgbd, scene_idx
+
+
+class ConcatIterable:
+    """
+    ConcatIterable is used to group iterable object.
+    When user iterate on this object, we will sample random iterable and return their
+    item with coresponding output_key.
+    With repeat_factors, user can do upsampling or downsampling to the iterables.
+    We mainly use this class to concat different data loader during training.
+
+    Args:
+        iterables: the iterable objects that will be grouped
+        output_keys: List of keys that is used to identify the iterable output.
+            The list length should be the same as number of iterables.
+        repeat_factors: List of numbers that represent the upsampling / downsampling factor
+            to the coresponding iterables. Should have same length as iterables.
+        seed: the seed for randomness
+    """
+
+    def __init__(self, iterables, output_keys, repeat_factors, seed=42):
+        self.iterables = iterables
+        self.output_keys = output_keys
+        self.repeat_factors = repeat_factors
+        self.seed = seed
+        self.num_iterables = len(self.iterables)
+        assert self.num_iterables == len(output_keys)
+        assert self.num_iterables == len(repeat_factors)
+
+        # The iterator len is adjusted with repeat_factors
+        self.iterator_lens = [
+            int(repeat_factors[i] * len(itb)) for i, itb in enumerate(self.iterables)
+        ]
+        self.max_total_steps = sum(self.iterator_lens)
+        self.indices = None
+        self.iterators = None
+
+        # self.step_counter == None indicate that self.indices are not yet initialized
+        self.step_counter = None
```

Comment:
In the current design on this class, you need to call `init_indices` for each epoch, or otherwise it will just return empty iterables. Example to ilustrate:
```
data_loaders = ConcatIterable(...)
data_loaders.init_indices()

for data in data_loaders:
  # Behave correctly
  ...
  
# Now data_loaders is empty and need to call `init_indices` before usage
for data in data_loaders:
  # Will terminate directly since now data_loaders is empty
  ...
```

If we call `init_indices` on the constructor, then user will often not aware of this behaviour because they get correct behaviour on the first epoch but not on the second epoch.